dot/peerset/peerstate.go

```diff
@@ -171,78 +169,111 @@ func (ps *PeersState) peerStatus(set int, peerID peer.ID) string {
 
 // peers return the list of all the peers we know of.
 func (ps *PeersState) peers() []peer.ID {
+	ps.peerStateRWMutex.RLock()
+	defer ps.peerStateRWMutex.RUnlock()
+
 	peerIDs := make([]peer.ID, 0, len(ps.nodes))
 	for k := range ps.nodes {
 		peerIDs = append(peerIDs, k)
 	}
 	return peerIDs
 }
 
+type connectedPeerReputation struct {
+	peerID     peer.ID
+	reputation Reputation
+}
+
```

Comment:
it looks like this is only used in the `sortedPeers` function, can you define it in there?